src/cache/externalResponseCacheService.ts

```diff
@@ -122,7 +122,7 @@ export class ExternalResponseCacheService {
         dataType: 'Map',
         value: Array.from(value.entries()), // or with spread: value: [...value]
       };
-    } else if (this[key] instanceof ethers.BigNumber) {
+    } else if (this[key] instanceof ethers.BigNumber || typeof this[key] === 'bigint') {
```

Comment:
Added this condition to check if the value is a bigint since JSON throws the error `do not know how to serialize a BigInt` for BigInt values